src/containers/CreatePipelineRun/YAMLEditor.test.js

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2019-2021 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+
+import { fireEvent, waitFor } from '@testing-library/react';
+import { renderWithRouter } from '../../utils/test';
+import { CreateYAMLEditor } from './YAMLEditor';
+import * as PipelineRunsAPI from '../../api/pipelineRuns';
+
+const submitButton = allByText => allByText('Create')[0];
+const pipelineRun = `
+      apiVersion: tekton.dev/v1beta1
+      kind: PipelineRun
+      metadata:
+        name: test-pipeline-run-name
+        namespace: test-namespace
+      spec:
+        pipelineSpec:
+          tasks:
+            - name: hello
+              taskSpec:
+                steps:
+                  - name: echo
+                    image: busybox
+                    script: |
+                      #!/bin/ash
+                      echo "Hello World!"
+    `;
+
+const pipelineRunWithoutNamespace = `
+      apiVersion: tekton.dev/v1beta1
+      kind: PipelineRun
+      metadata:
+        name: test-pipeline-run-name
+      spec:
+        pipelineSpec:
+          tasks:
+            - name: hello
+              taskSpec:
+                steps:
+                  - name: echo
+                    image: busybox
+                    script: |
+                      #!/bin/ash
+                      echo "Hello World!"
+    `;
+
+const pipelineRunIncorrectYaml = `a: b
+dddd;a`;
+
+const pipelineRunRaw = {
+  apiVersion: 'tekton.dev/v1beta1',
+  kind: 'PipelineRun',
+  metadata: { name: 'test-pipeline-run-name', namespace: 'test-namespace' },
+  spec: {
+    pipelineSpec: {
+      tasks: [
+        {
+          name: 'hello',
+          taskSpec: {
+            steps: [
+              {
+                image: 'busybox',
+                name: 'echo',
+                script: '#!/bin/ash\necho "Hello World!"\n'
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+};
+
+describe('YAMLEditor', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.spyOn(window.history, 'pushState');
+    // WA for codemirror vs jsdom https://github.com/jsdom/jsdom/issues/3002#issuecomment-1118039915
```

Comment:
What is 'WA' in this context? Guessing workaround…
If that's the case let's just go with the full word to avoid confusion.